### PR TITLE
Backend functionality for card deletion was implemented

### PR DESCRIPTION
**Pull request by KuryloVolodymyr**
Backend functionality for card deletion was implemented

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java [hunk 1]
```diff
@@ -50,7 +50,16 @@ public void updateCard(Long id, Card card) {
         return cardsQueue;
     }
 
-    public void deleteCard(Long id) {
-        cardRepository.delete(id);
+    public void deleteCard(Long deck_id, Long card_id) {
+        Deck deck = deckRepository.getDeckById(deck_id);
+        List<Card> cards = deck.getCards();
+        for(Card card: cards) {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: Why not getting the card by id directly?

Card cardToDelete = cardRepository.getCardById(...)

It wrong to iterate over all the cards to find one.

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java [hunk 2]
```diff
@@ -50,7 +50,16 @@ public void updateCard(Long id, Card card) {
         return cardsQueue;
     }
 
-    public void deleteCard(Long id) {
-        cardRepository.delete(id);
+    public void deleteCard(Long deck_id, Long card_id) {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: Please rename parameters https://sbforge.org/sonar/rules/show/squid:S00117

Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: Does it work without @ Transactional annotation?

Comment (KuryloVolodymyr) [reply src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: Yes, it works without @ Transactional annotation

**Pull request closed without merging**